### PR TITLE
Enable host dataset for IVF-Flat

### PR DESCRIPTION
**Pull request by tfeher**
Enable host input data for IVF-Flat build.

This is done by batch-wise processing the dataset during extend, similarly how IVF-PQ does it.

### CODE REVIEW

File: cpp/include/raft/neighbors/detail/ivf_flat_build.cuh [hunk 1]
```diff
@@ -335,12 +403,18 @@ inline auto build(raft::resources const& handle,
     auto centers_view = raft::make_device_matrix_view<float, IdxT>(
       index.centers().data_handle(), index.n_lists(), index.dim());
     raft::cluster::kmeans_balanced_params kmeans_params;
+    RAFT_CUDA_TRY(cudaMemGetInfo(&free_mem, &total_mem));
+    RAFT_LOG_INFO("ivf_flat::build allocated mem %6.1f GiB after trainset copied",
```
[the remark this answers is not in the document]
Comment (tfeher) [reply cpp/include/raft/neighbors/detail/ivf_flat_build.cuh]: No, it was used for debugging only, I have removed these.

**Comment by tfeher** [comment]
Squashed the PR to a single commit to remove unsigned ones.

**Review by cjnolet — APPROVED** [review]
LGTM!

**Pull request merged**